teuthology/orchestra/daemon/cephdaemonunit.py

```diff
@@ -0,0 +1,201 @@
+import logging
+import re
+from cStringIO import StringIO
+
+from teuthology.exceptions import CommandFailedError
+from teuthology.orchestra import run
+from teuthology.orchestra.daemon.state import DaemonState
+
+log = logging.getLogger(__name__)
+
+class CephDaemonUnit(DaemonState):
+    def __init__(self, remote, role, id_, *command_args,
+                 **command_kwargs):
+        super(CephDaemonUnit, self).__init__(
+            remote, role, id_, *command_args, **command_kwargs)
+        self._set_commands()
+        self.log = command_kwargs.get('logger', log)
+        self.is_started = command_kwargs.get('started', False)
+
+    def name(self):
+        return '%s.%s' % (self.type_, self.id_)
+
+    def _get_systemd_cmd(self, action):
+        return [
+            'sudo', 'systemctl',
+            action,
+            'ceph-%s@%s.%s' % (self.fsid, self.type_, self.id_),
+        ]
+
+    def _set_commands(self):
+        self.start_cmd = self._get_systemd_cmd('start')
+        self.stop_cmd = self._get_systemd_cmd('stop')
+        self.restart_cmd = self._get_systemd_cmd('restart')
+        self.show_cmd = self._get_systemd_cmd('show')
+        self.status_cmd = self._get_systemd_cmd('status')
+
+    def check_status(self):
+        """
+        Check to see if the process has exited.
+
+        :returns: The exit status, if any
+        :raises:  CommandFailedError, if the process was run with
+                  check_status=True
+        """
+        raise NotImplementedError
+        proc = self.remote.run(
```

Comment:
Could you please `remote.sh` instead of `remote.run` in order to make easy to make the code py2/py3 compatible:
```output = self.remote.sh(self.show_cmd + ' | grep -i state')```